### PR TITLE
vmm: Replace all use of RwLock with Mutex

### PR DESCRIPTION
**Pull request by crab2313**
Fixes: #777

Signed-off-by: Qiu Wenbo <qiuwenbo@phytium.com.cn>

### CODE REVIEW

**Comment by sboeuf** [comment]
Thanks for the PR @crab2313 :)

**Comment by sboeuf** [comment]
You're hitting the 1 hour timeout on the CI. This means you might have a deadlock somewhere.

File: vm-virtio/src/iommu.rs [hunk 1]
```diff
@@ -759,16 +759,16 @@ struct Mapping {
...
+    mappings: Arc<Mutex<BTreeMap<u32, BTreeMap<u64, Mapping>>>>,
 }
 
 impl DmaRemapping for IommuMapping {
     fn translate(&self, id: u32, addr: u64) -> std::result::Result<u64, std::io::Error> {
         debug!("Translate addr 0x{:x}", addr);
-        if let Some(domain) = self.endpoints.read().unwrap().get(&id) {
-            if let Some(mapping) = self.mappings.read().unwrap().get(domain) {
+        if let Some(domain) = self.endpoints.lock().unwrap().get(&id) {
+            if let Some(mapping) = self.mappings.lock().unwrap().get(domain) {
```
Comment (rbradford) [inline vm-virtio/src/iommu.rs]: I *think* the deadlock is here (just from inspection.)

Comment (crab2313) [reply vm-virtio/src/iommu.rs]: Thank you for the hint.

**Comment by crab2313** [comment]
It seems more complicated than that. I just setup the integration test environment locally and find that almost every single replacement of a lock may cause a hang. I will reopen the pull request when I figure them out.

**Pull request closed without merging**